Jellyfin.Plugin.AniDB/Providers/equals_check.cs

```diff
@@ -20,528 +18,217 @@ public Equals_check(ILogger<Equals_check> logger)
+                            lowestDistance = stringDistance;
+                            currentId = id;
                         }
                     }
-                    x++;
-                }
-                if (pre_aid.Count == 1)
-                {
-                    if (!string.IsNullOrEmpty(pre_aid[0]))
-                    {
-                        return pre_aid[0];
-                    }
-                }
-                int biggestcount = 0;
-                string cache_aid="";
-                if (a == b)
-                {
-                    foreach (string _aid in pre_aid)
-                    {
-                       string result= await One_line_regex(new Regex(@"<anime aid=" + "\"" + _aid + "\"" + @"((?s).*?)<\/anime>"), xml, cancellationToken);
-                       int count = (result.Length - result.Replace(a, "").Length)/a.Length;
-                       if(biggestcount< count)
-                       {
-                            biggestcount = count;
-                            cache_aid =_aid;
-                       }
-                    }
-                    if (!string.IsNullOrEmpty(cache_aid))
-                    {
-                        return cache_aid;
-                    }
-                }
-                foreach (string _aid in pre_aid)
-                {
-                    XElement doc = XElement.Parse("<?xml version=\"1.0\" encoding=\"UTF - 8\"?>" + "<animetitles>" +await One_line_regex(new Regex("<anime aid=\"" + _aid + "\">" + @"(?s)(.*?)<\/anime>"), xml, cancellationToken,0, 0) + "</animetitles>");
-                    var a_ = from page in doc.Elements("anime")
-                             where _aid == page.Attribute("aid").Value
-                             select page;
-
-                    if (await Simple_compare(a_.Elements("title"), b, cancellationToken) && await Simple_compare(a_.Elements("title"), a, cancellationToken))
-                    {
-                        return _aid;
-                    }
-                }
-                return "";
-            }
-            catch (Exception)
-            {
-                if (x_ == 1)
-                {
-                    return "";
-                }
-                else
-                {
-                    await Task.Run(() => AniDbTitleDownloader.Load_static(cancellationToken), cancellationToken);
-                    return await Fast_xml_search(a, b, cancellationToken, true, 1);
                 }
             }
+            return currentId;
         }
 
         /// <summary>
-        /// get file Path from anidb xml file
+        /// Calculates the Levenshtein distance - a metric for measuring the difference between two strings.
+        /// The higher the number, the more different the two strings are.
         /// </summary>
-        /// <returns></returns>
-        private static string Get_anidb_xml_file()
+        private static int LevenshteinDistance(string str1, string str2)
```

Comment:
:heart: 
*while I love this, I hope there are not too many performance implications by it for larger libraries

Reply:
I thought about that as well. I tested it on my pretty large library and compared to other things running while scanning you don't notice it. You usually have to wait for image extraction and the aniDB api.